streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/SubscriptionInfo.java

```diff
@@ -123,7 +123,7 @@ public ByteBuffer encode() {
                 break;
             case 3:
             case 4:
-                buf = encodeVersionThree();
+                buf = encodeVersionThree(usedVersion);
```

Comment:
It is simpler to just have an `encodeVersionFour` which does not change anything than `encodeVersionThree` but just put the different version. Note that within this release cycle we may introduce other format changes as well (so eventually the `encodeVersionFour` may be implemented differently anyways).